apps/remote_control/lib/lexical/remote_control/code_action/handlers/organize_aliases.ex

```diff
@@ -0,0 +1,173 @@
+            "alias #{join(a.module)}, as: #{join(List.wrap(a.as))}"
+          end
+
+        indent(text, initial_spaces)
+      end)
+      |> String.trim_trailing()
+
+    zeroed_start = %Position{first_alias_start | character: 1}
+    new_alias_range = Range.new(zeroed_start, zeroed_start)
+    edits = remove_old_aliases(aliases)
+
+    edits ++
+      [Edit.new(alias_text, new_alias_range)]
+  end
+
+  defp remove_old_aliases(aliases) do
+    ranges =
+      aliases
+      # iterating back to start means we won't have prior edits
+      # clobber subsequent edits
+      |> Enum.sort_by(& &1.range.start.line, :desc)
+      |> Enum.uniq_by(& &1.range)
+      |> Enum.map(fn %Alias{} = alias ->
+        orig_range = alias.range
+
+        orig_range
+        |> put_in([:start, :character], 1)
+        |> update_in([:end], fn %Position{} = pos ->
+          %Position{pos | character: 1, line: pos.line + 1}
+        end)
+      end)
+
+    first_alias_index = length(ranges) - 1
+
+    ranges
+    |> Enum.with_index()
+    |> Enum.map(fn
+      {range, ^first_alias_index} ->
+        # add a new line where the first alias was to make space
+        # for the rewritten aliases
+        Edit.new("\n", range)
+
+      {range, _} ->
+        Edit.new("", range)
+    end)
+  end
+
+  defp check_aliases(%Document{}, %Analysis{} = analysis, %Range{} = range) do
+    narroest_scope =
+      analysis
+      |> Analysis.scopes_at(range.start)
+      |> narrorwest_scope(range.start)
+
+    with %Scope{} <- narroest_scope,
+         false <- Enum.empty?(narroest_scope.aliases) do
+      :ok
+    else
+      _ ->
+        :error
+    end
+  end
+
+  defp aliases_in_scope(%Scope{} = scope) do
+    scope.aliases
+    |> Enum.filter(fn %Alias{} = scope_alias ->
+      scope_alias.explicit? and Range.contains?(scope.range, scope_alias.range.start)
+    end)
+    |> Enum.sort_by(fn %Alias{} = scope_alias ->
+      Enum.map(scope_alias.module, fn elem -> elem |> Atom.to_string() |> String.downcase() end)
+    end)
+  end
+
+  defp aliases_in_scope(_) do
+    []
+  end
+
+  defp enclosing_scopes(scopes, range) do
+    Enum.filter(scopes, fn scope ->
+      Range.contains?(scope.range, range.start)
+    end)
+  end
+
+  defp first_alias_range(aliases) do
+    aliases
+    |> Enum.min_by(fn %Alias{} = a ->
+      {a.range.start.line, a.range.start.character}
+    end)
+    |> Map.get(:range)
+  end
+
+  defp join(module) do
+    Enum.join(module, ".")
+  end
+
+  defp indent(text, spaces) do
+    String.duplicate(" ", spaces) <> text
+  end
+
+  defp narrorwest_scope(scope_list, %Position{} = position) do
+    Enum.reduce(scope_list, nil, fn
+      scope, nil ->
+        scope
+
+      %Scope{id: :global}, %Scope{} = current ->
+        current
+
+      %Scope{} = next_scope, %Scope{} = current_scope ->
+        Enum.min_by([next_scope, current_scope], fn %Scope{} = scope ->
+          scope_start = scope.range.start
+          position.line - scope_start.line
```

Comment:
It looks like you are trying to find the scope closest to the cursor position_line. If that's the case, why not name the function  `nearest_scope`?

Reply:
That was the initial name, my thinking is that "near" has a couple meanings, for example
```elixir
SCOPE_1 do 
end
SCOPE_2 do 
end
```
Both SCOPE_1 and SCOPE_2 are near each other, but what I want to express is that
```elixir
SCOPE_1 do
  SCOPE_2 do 
  end
  SCOPE_3 do
  end
end
```
SCOPE_1 is the enclosing scope that's closest, and maybe that's the best name `narrowest_enclosing_scope`

Reply:
Sure